drivers/cas_ce_data/app/models/cas_ce_data/synthetic/event.rb

```diff
@@ -43,7 +43,12 @@ def self.remove_orphans
     end
 
     def self.add_new
-      new_events = CasCeData::GrdaWarehouse::CasReferralEvent.where.not(id: self.select(:source_id))
+      # Replaced the following with a lateral join for performance, leaving temporarily for easy reversal
+      # new_events = CasCeData::GrdaWarehouse::CasReferralEvent.where.not(id: self.select(:source_id))
+      existing = where('"cas_referral_events"."id" = source_id').select(:source_id).limit(1)
```

Comment:
I saw in the real database that this table had a type that was all the same value. Not sure if you need to account for that in other installations.

Reply:
Does AR get the `source_type` right? I'm worried about coincidentally overapping `source_ids`?